pkg/controller/servicebinding/secret.go

```diff
@@ -48,36 +127,21 @@ func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1
 
 	resourceClient := s.buildResourceClient()
 
-	logger.Debug("Attempt to create secret...")
 	existingSecret, err := s.get()
 	if err != nil {
 		if errors.IsNotFound(err) {
-			_, err := resourceClient.Create(u, metav1.CreateOptions{})
+			logger.Info("Secret not found. Creating secret")
+			secretNew, err := resourceClient.Create(u, metav1.CreateOptions{})
 			if err != nil {
 				logger.Error(err, "Error creating secret")
 				return nil, err
 			}
 			logger.Info("Secret created")
-			return u, nil
+			return secretNew, nil
 		}
 		return nil, err
 	}
-	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
-	payloadInterim := make(map[string]interface{})
-	for k, v := range payload {
-		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
-	}
-	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
-	if comparisonResult.Success {
-		logger.Debug("Secret data is same. Skip Update")
-	} else {
-		logger.Info("Secret data is different; update secret", "Diff", comparisonResult.Diff)
-		_, err = resourceClient.Update(u, metav1.UpdateOptions{})
-		if err != nil {
-			return nil, err
-		}
-	}
-	return u, nil
+	return existingSecret, nil
```

Comment:
It seems the secret is being read from the cluster and not validated against what is expected it to contain; I understand we do observe secrets to reconcile the Service Binding if its secret is changed, but one needs to keep in mind the operator doesn't prohibit (as of yet) the secret to be written, but it does update the secret with the proper content once a change is identified, and there's a window of time in-between.

Returning an error in the case the retrieved data's hash is different than the expected hash contained in the resource name is expected here as well.

Reply:
> Returning an error in the case the retrieved data's hash is different than the expected hash contained in the resource name is expected here as well.

I check name and contained data here:

https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-0141e5e3383464862fa2299425f3318875159d89221a6cd2f36088f6d06d75b1R108

Reply:
Oh, I entirely missed that, sorry for that! Thanks!